sw/device/tests/i2c_host_hdc1080_humidity_temp_test.c

```diff
@@ -0,0 +1,141 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+#include <assert.h>
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/macros.h"
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_i2c.h"
+#include "sw/device/lib/runtime/hart.h"
+#include "sw/device/lib/runtime/irq.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/runtime/print.h"
+#include "sw/device/lib/testing/i2c_testutils.h"
+#include "sw/device/lib/testing/rv_core_ibex_testutils.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "i2c_regs.h"  // Generated.
+
+OTTF_DEFINE_TEST_CONFIG();
+
+enum {
+  kDeviceAddr = 0x40,
+
+  // Registers
+  kTemperatureReg = 0x00,
+  kHumidityReg = 0x01,
+  kConfigurationReg = 0x02,
+  kManufacturerIdReg = 0xFE,
+  kDeviceIdReg = 0xFF,
+
+  // Registers values
+  kManufacturerId = 0x5449,
+  kDeviceId = 0x1050,
+
+  // Other
+  kDefaultTimeoutMicros = 5000,
+};
+
+static dif_rv_core_ibex_t rv_core_ibex;
+static dif_pinmux_t pinmux;
+static dif_i2c_t i2c;
+
+static status_t read_manufacture_id(void) {
+  uint8_t reg = kManufacturerIdReg;
+  uint8_t data[2] = {0};
+  TRY(i2c_testutils_write(&i2c, kDeviceAddr, 1, &reg, true));
+  TRY(i2c_testutils_read(&i2c, kDeviceAddr, 2, data, kDefaultTimeoutMicros));
+  uint8_t manufacturer_id[2] = {kManufacturerId >> 8, kManufacturerId & 0xFF};
+  TRY_CHECK_ARRAYS_EQ(data, manufacturer_id, sizeof(data));
+  return OK_STATUS();
+}
+
+static status_t read_product_id(void) {
+  uint8_t reg = kDeviceIdReg;
+  uint8_t data[2] = {0};
+  TRY(i2c_testutils_write(&i2c, kDeviceAddr, 1, &reg, true));
+  TRY(i2c_testutils_read(&i2c, kDeviceAddr, 2, data, kDefaultTimeoutMicros));
+  uint8_t device_id[2] = {kDeviceId >> 8, kDeviceId & 0xFF};
+  TRY_CHECK_ARRAYS_EQ(data, device_id, sizeof(data));
+  return OK_STATUS();
+}
+
+static status_t read_temperature(void) {
+  uint8_t reg = kTemperatureReg;
+  uint8_t data[2] = {0};
+  TRY(i2c_testutils_write(&i2c, kDeviceAddr, 1, &reg, true));
+  busy_spin_micros(100000);
```

Comment:
Drop this delays as we have a retry with timeout inside the read function below.